model_signing/hashing/model_hash.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This package provides functionality to hash the model folder and create a statement."""
+from typing import Optional
+
+import hashlib
+import os
+import pathlib
+
+from concurrent import futures
+
+from in_toto_attestation.v1 import statement 
+from in_toto_attestation.v1 import resource_descriptor as res_desc
+
+
+_CHUNK_SIZE = int(1_000_000_000) # ~1GB
+_MAX_WORKERS = 10
```

Comment:
do we need to specify max workers? I think python has a threshold set by default, so we can let the runtime decide?